### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by spbnick**
This introduces a host of minor fixes and improvements done during the work on table partitioning and query time-limiting.

### CODE REVIEW

**Comment by spbnick** [comment]
@mrbazzan, do you think you could take a stab at reviewing these?

**Comment by mrbazzan** [comment]
> @mrbazzan, do you think you could take a stab at reviewing these?

Sure. Anything you want me to especially look out for?

**Comment by spbnick** [comment]
> > @mrbazzan, do you think you could take a stab at reviewing these?
> 
> Sure. Anything you want me to especially look out for?

No, just please try to find anything wrong with it, if you can. Thank you :relaxed:

File: kcidb/db/sqlite/schema.py [hunk 1]
```diff
@@ -84,21 +97,32 @@ def __init__(self, constraint=None, pack=None, unpack=None):
...
                             A member of the Constraint enum, or None,
                             meaning no constraint.
-            pack:           The function packing the JSON representation of
-                            column's values into the SQLite representation.
-            unpack:         The function unpacking the SQLite representation
-                            of column's values into the JSON representation.
         """
         assert constraint is None or isinstance(constraint, Constraint)
         assert pack is None or callable(pack)
         assert unpack is None or callable(unpack)
```
Comment (mrbazzan) [inline kcidb/db/sqlite/schema.py:104]: Shouldn't `pack` and `unpack` parameters be removed from the constructor? I also think the validation statement should be removed.

File: kcidb/misc.py [hunk 2]
```diff
@@ -194,6 +195,32 @@ def non_negative_int_or_inf(string):
...
+def iso_timestamp(string):
+    """
+    Parse an ISO-8601 timestamp out of a string, assuming local timezone, if
+    not specified. Matches the argparse type function interface.
+
+    Args:
+        string: The string to parse.
+
+    Returns:
+        The non-negative integer, or  parsed out of the string.
```
Comment (mrbazzan) [inline kcidb/misc.py:207]: Incoherent `Returns:`

Should be something like: `The timestamp parsed out of the string.`

**Review by mrbazzan — CHANGES_REQUESTED** [review]

**Comment by mrbazzan** [comment]
> @mrbazzan, do you think you could take a stab at reviewing these?

Looks good aside the two suggested changes.

**Comment by spbnick** [comment]
Thanks a lot, @mrbazzan, both are good catches! Should be fixed now, please check :relaxed:

**Comment by mrbazzan** [comment]
@spbnick LGTM!

**Comment by spbnick** [comment]
Thanks, Abdulwasiu, merging!

**Pull request merged**